libs/ngxtension/inject-local-storage/src/inject-local-storage.spec.ts

```diff
@@ -29,6 +29,16 @@ describe('injectLocalStorage', () => {
 			});
 		}));
 
+		it('should return data of defaultValue', () => {
+			TestBed.runInInjectionContext(() => {
+				const defaultValue = 'default';
+				const localStorageSignal = injectLocalStorage<string>(key, {
```

Comment:
Can we include a test that only checks for the type and not just the value?

Reply:
It can be difficult to differentiate between an exact type union and undefined (e.g., string | undefined). At runtime, we typically just receive a string. I improved the tests to cover cases where a default value is provided (ensuring we don't return undefined) and cases without a default value, where undefined is returned from local storage. Please verify it once again and let me know if it works for you. Additionally, I think those tests document the behavior I explained before. Nevertheless, I'd add appropriate info in docs as well.